.github/workflows/CI.yml

```diff
@@ -36,21 +40,19 @@ jobs:
     - name: Generate coverage report
       run: |
         PYTHONPATH=src
-        pytest --cov=./src tests/
+        pytest --cov=./src tests/ --cov-report term --cov-report=xml:coverage.xml
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v3
       with:
         verbose: true
-    - name: Install pypa/build
-      run: >-
-        python -m
-        pip install
-        build
-        --user
-    - name: Build a binary wheel and a source tarball
-      run: >-
-        python -m
-        build
-        --sdist
-        --wheel
-        --outdir dist/
+        directory: ./coverage/reports/
+        env_vars: OS,PYTHON
+        fail_ci_if_error: true
+        files: ./coverage.xml
+        flags: unittests
+        name: codecov-umbrella
+#        token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
You are not adding tokens? Did you forget to uncomment?

Reply:
For public repositories, no token is needed. 
Source: https://github.com/codecov/codecov-action

I am removing this line.